### PR TITLE
Change address creation

### PR DESCRIPTION
**Pull request by dolgopolovwork**
1) `btc-pregen` is renamed to 'btc-address-generation', because it's used to generate different types of addresses
2) Now we can generate "change" address, that will be used as an address to collect changes

### CODE REVIEW

File: notary-integration-test/src/integration-test/kotlin/integration/btc/BtcRegistrationIntegrationTest.kt [hunk 1]
```diff
@@ -151,7 +149,26 @@ class BtcRegistrationIntegrationTest {
...
+     * @when client name is passed to registration service
+     * @then client stays unregistered, because there are no "free" Bitcoin addresses
+     */
+    @Test
+    fun testRegistrationOnlyChangeAddresses() {
+        val clientsBeforeRegistration = btcTakenAddressesProvider.getRegisteredAddresses().get().size
+        integrationHelper.genChangeBtcAddress(btcNotaryConfig.bitcoin.walletPath)
+        val keypair = ModelCrypto().generateKeypair()
+        val userName = String.getRandomString(9)
+        val res = khttp.post(
```
Comment (grimadas) [inline notary-integration-test/src/integration-test/kotlin/integration/btc/BtcRegistrationIntegrationTest.kt]: Is there another way to register user?

Comment (dolgopolovwork) [reply notary-integration-test/src/integration-test/kotlin/integration/btc/BtcRegistrationIntegrationTest.kt]: Yes. We can do it directly from Iroha.

File: btc-address-generation/src/main/kotlin/generation/btc/trigger/AddressGenerationExecutor.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import provider.btc.BtcSessionProvider
+import provider.btc.address.BtcAddressType
+import sidechain.iroha.IrohaInitialization
+import sidechain.iroha.consumer.IrohaNetwork
+import sidechain.iroha.consumer.IrohaNetworkImpl
+import sidechain.iroha.util.ModelUtil
+
+private val logger = KLogging().logger
+
+//TODO springify this file
```
Comment (grimadas) [inline btc-address-generation/src/main/kotlin/generation/btc/trigger/AddressGenerationExecutor.kt]: Add issue in jira

Comment (dolgopolovwork) [reply btc-address-generation/src/main/kotlin/generation/btc/trigger/AddressGenerationExecutor.kt]: [Done](https://soramitsu.atlassian.net/browse/D3-622)

File: btc-withdrawal/src/main/kotlin/withdrawal/btc/provider/BtcChangeAddressProvider.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+@Component
+class BtcChangeAddressProvider(
+    @Autowired private val btcAddressesProvider: BtcAddressesProvider,
+    @Autowired private val btcNetworkConfigProvider: BtcNetworkConfigProvider
+) {
+    /**
+     * Returns change address
+     * @return - result with change address object
+     */
+    fun getChangeAddress(): Result<Address, Exception> {
```
[the remark this answers is not in the document]
Comment (dolgopolovwork) [reply btc-withdrawal/src/main/kotlin/withdrawal/btc/provider/BtcChangeAddressProvider.kt]: Do you want me to cache this value?

File: btc/src/main/kotlin/provider/btc/address/BtcRegisteredAddressesProvider.kt [hunk 4]
```diff
@@ -18,7 +18,7 @@ open class BtcRegisteredAddressesProvider(
      * Get all registered btc addresses
      * @return list full of registered BTC addresses
      */
-    fun getRegisteredAddresses(): Result<List<BtcAddress>, Exception> {
+    fun getRegisteredAddresses(): Result<MutableList<BtcAddress>, Exception> {
```
Comment (grimadas) [inline btc/src/main/kotlin/provider/btc/address/BtcRegisteredAddressesProvider.kt]: Put it to other place

**Pull request merged**